apps/req.c

```diff
@@ -1641,7 +1564,22 @@ static EVP_PKEY_CTX *set_keygen_ctx(const char *gstr,
         EVP_PKEY_CTX_free(gctx);
         return NULL;
     }
-    if ((*pkey_type == EVP_PKEY_RSA) && (keylen != -1)) {
+    if ((EVP_PKEY_CTX_is_a(gctx, "RSA") || EVP_PKEY_CTX_is_a(gctx, "RSA-PSS"))
```

Comment:
In the original code there was also handling of DSA for MIN_KEY_LENGTH and OPENSSL_DSA_MAX_MODULUS_BITS, can you please put it back?

Reply:
Yeah, it felt like a dead code, but it isn't. Sorry for that.